### PR TITLE
feat: wait a full feedback update to write it

### PR DESCRIPTION
**Pull request by seixasxbr**
Depends on:
- [ ] https://github.com/tidewise/drivers-bms_victron_smart_shunt/pull/2/

On top of: #1 

# What is this PR for
Write data struct only when it has a full update.
The device sends the data divided in two packets, so it's necessary to merge the data to compose the feedback struct.

# Results, How I tested
BATTERY STATUS
image

MERGE
https://github.com/user-attachments/assets/cc8fffbd-dee3-4651-b350-15759e01c15b

# Checklist
- [x] Assign yourself  in the PR

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -56,9 +55,33 @@ void bms_victron_smart_shunt::Task::cleanupHook()
...
+static BatteryStatus toBatteryStatus(SmartShuntFeedback const& feedback,
+    double max_current)
+{
+    BatteryStatus battery_status;
+    battery_status.time = feedback.timestamp;
+    battery_status.temperature = feedback.temperature;
+    battery_status.charge = feedback.state_of_charge / 100;
+    battery_status.current = feedback.current;
+    battery_status.voltage = feedback.voltage;
+    battery_status.max_current = max_current;
```
Comment (jhonasiv) [inline tasks/Task.cpp:67]: I dont think that max current is the same thing you are doing here. To me, it seems that max `battery_status.max_current` should come from the sensor itself, but I might be wrong.

Comment (higor-melo) [reply tasks/Task.cpp:67]: Isn't he updating the m_max_current according to the feedback current of each loop? It seems like the sensor doesn't send a max_current value and this is a workaround to store this value.

**Review by higor-melo — APPROVED** [review]

File: tasks/Task.cpp (hunk 1 above)
Comment (jhonasiv) [reply tasks/Task.cpp:67]: Yeah, my point is that the work around is misinformation, its better to not send anything in that field

Comment (seixasxbr) [reply tasks/Task.cpp:67]: Already talked with sylvain about that. He agrees to hold the max current value in the task

Comment (jhonasiv) [reply tasks/Task.cpp:67]: Fair enough

**Review by jhonasiv — APPROVED** [review]

**Pull request merged**